openedx/core/djangoapps/xblock/runtime/ephemeral_field_data.py

```diff
@@ -0,0 +1,62 @@
+"""
+An :class:`~xblock.runtime.KeyValueStore` that stores data in the django cache
+
+This is used for low-priority ephemeral student state data:
+* Anonymous users browsing and previewing content
+* Studio authors testing out XBlocks
+
+We could also store this data in django sessions, but its a bit tricky to access
+session data during any requests which don't have any cookies or other normal
+authentication mechanisms (like XBlock handler calls from within XBlock <iframe>
+sandboxes). And keeping this storage completely separate from django session
+data and registered user XBlock state reduces the potential for security
+problems. We expect the data in this store to be low-value and free of
+personally identifiable information (PII) so if some security bug results in one
+user accessing a different user's entries in this particular store, it's not a
+big deal.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from django.conf import settings
+from django.core.cache import caches
+from xblock.runtime import KeyValueStore
+
+
+FIELD_DATA_TIMEOUT = 2 * 24 * 60 * 60  # expires after 2 days
```

Comment:
Does it need a timeout? It'll just cycle off the end of the cache, no?

Reply:
From what I can tell with a quick look, the default LMS cache doesn't specify a default expiry time, so [the Django default of 300 seconds](https://docs.djangoproject.com/en/3.0/topics/cache/#cache-arguments) would be used, which is too short.

Reply:
I could set the timeout to `None` though (keep forever), which is simpler? I think that would behave the way you're suggesting.

Reply:
Yeah. It's not something I feel strong enough to block on, just a suggestion.